src/Symfony/Component/HttpClient/HttpClientTrait.php

```diff
@@ -71,8 +71,17 @@ private static function prepareRequest(?string $method, ?string $url, array $opt
             throw new InvalidArgumentException(sprintf('Option "on_progress" must be callable, %s given.', \is_object($onProgress) ? \get_class($onProgress) : \gettype($onProgress)));
         }
 
+        if (\is_array($options['auth'] ?? null)) {
+            $count = \count($options['auth']);
+            if ($count <= 0 || $count > 2) {
```

Comment:
`if (!\in_array(\count($options['auth']), [1, 2], true))`?

Reply:
As you wish!

Reply:
changed

Reply:
TBH, I prefer the more explicit way.

Reply:
reverted :)